### PR TITLE
fix: refactor to config from executor settings

### PR DESCRIPTION
**Pull request by jakevc**
- Fixed env vars in tests
- Removed AzBatchConfig class, using ExecutorSettings in it's place
- Removed unused funcs, and other cleanup

### CODE REVIEW

File: snakemake_executor_plugin_azure_batch/__init__.py [hunk 1]
```diff
@@ -176,6 +173,14 @@ class ExecutorSettings(ExecutorSettingsBase):
             "env_var": False,
         },
     )
+    container_image: Optional[str] = field(
```
Comment (johanneskoester) [inline snakemake_executor_plugin_azure_batch/__init__.py]: container_image is a global snakemake option. The executor does not need its own: https://github.com/snakemake/snakemake/blob/aa1c2839165fc03d20e5a28657b2e1fa7ffdf80e/snakemake/settings.py#L349

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Comment by jakevc** [comment]
Is it obvious to you why this is failing @johanneskoester ? I am having a hard time tracking down a fix.. https://github.com/snakemake/snakemake-executor-plugin-azure-batch/actions/runs/7117767600/job/19379237614#step:8:102

Seems like it's trying to update the job in postprocess after the job is done

**Comment by johanneskoester** [comment]
> Is it obvious to you why this is failing @johanneskoester ? I am having a hard time tracking down a fix.. https://github.com/snakemake/snakemake-executor-plugin-azure-batch/actions/runs/7117767600/job/19379237614#step:8:102
> 
> Seems like it's trying to update the job in postprocess after the job is done

This is suprising, the other executors don't have that problem. I have added a debug message to the main Snakemake branch, let's see whether a rerun with that sheds some light on this.

**Review by johanneskoester — APPROVED** [review]

**Comment by johanneskoester** [comment]
Yay, fixed! The problem was that you reported job success and error multiple times. THose have to be called only once per job.

**Comment by jakevc** [comment]
awesome, thanks

**Pull request merged**